compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -1598,9 +1604,10 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
         opaque_ty_node_id: NodeId,
         origin: hir::OpaqueTyOrigin,
         in_trait: bool,
-        captured_lifetimes_to_duplicate: Vec<Lifetime>,
+        captured_lifetimes_to_duplicate: FxIndexSet<Lifetime>,
```

Comment:
looping over this set in this function will lead to various ordering instabilities.

Reply:
This is an `FxIndexSet` though

Reply:
🤦 no more reviews by me after 10pm